{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4eb8bd31",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/neba/Desktop/CyclicaVsFDA/TEST_cyclica_FDA_working_11062022\r\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.chdir (\"/Users/neba/Desktop/CyclicaVsFDA/TEST_cyclica_FDA_working_11062022/\")\n",
    "os.getcwd()\n",
    "    \n",
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d18b4fd8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 98584\r\n",
      "-rw-r--r--   1 neba  staff  6931030  3 Jun 03:56 Untitled1-Copy1.ipynb\r\n",
      "-rw-r--r--   1 neba  staff    61782  3 Jun 04:46 Untitled1-Copy2.ipynb\r\n",
      "-rw-r--r--   1 neba  staff  6923827  3 Jun 04:14 Untitled1-Copy3.ipynb\r\n",
      "-rw-r--r--   1 neba  staff  6923831  3 Jun 03:38 Untitled1-Copy4.ipynb\r\n",
      "-rw-r--r--   1 neba  staff  6923831  3 Jun 04:16 Untitled1-Copy5.ipynb\r\n",
      "-rw-r--r--   1 neba  staff    62889  3 Jun 04:50 Untitled1-Copy6.ipynb\r\n",
      "-rw-r--r--@  1 neba  staff    10831 12 Jun 20:46 Untitled1.ipynb\r\n",
      "-rw-r--r--   1 neba  staff     4231 30 May 21:57 control_cyc1.pdb\r\n",
      "-rw-r--r--   1 neba  staff     6118 30 May 21:58 control_cyc2.pdb\r\n",
      "-rw-r--r--   1 neba  staff      636  2 Jun 18:01 control_fda1.pdb\r\n",
      "-rw-r--r--   1 neba  staff     4231 30 May 21:57 cyclica1.pdb\r\n",
      "-rw-r--r--   1 neba  staff     6118 30 May 21:58 cyclica2.pdb\r\n",
      "-rw-r--r--   1 neba  staff     1663  2 Jun 17:56 fda901.pdb\r\n",
      "-rw-r--r--   1 neba  staff     2058  2 Jun 17:56 fda902.pdb\r\n",
      "-rw-r--r--   1 neba  staff     2137  2 Jun 17:56 fda903.pdb\r\n",
      "-rw-r--r--   1 neba  staff     1821  2 Jun 17:56 fda904.pdb\r\n",
      "-rw-r--r--   1 neba  staff      636  2 Jun 17:56 fda905.pdb\r\n",
      "-rw-r--r--   1 neba  staff     2769  2 Jun 17:56 fda906.pdb\r\n",
      "-rw-r--r--   1 neba  staff      794  2 Jun 17:56 fda907.pdb\r\n",
      "-rw-r--r--   1 neba  staff     1189  2 Jun 17:56 fda908.pdb\r\n",
      "-rw-r--r--   1 neba  staff     1031  2 Jun 17:56 fda909.pdb\r\n",
      "-rw-r--r--   1 neba  staff     3401  2 Jun 17:56 fda910.pdb\r\n",
      "-rw-r--r--   1 neba  staff     2769  2 Jun 17:56 fda911.pdb\r\n",
      "-rw-r--r--   1 neba  staff     3243  2 Jun 17:56 fda912.pdb\r\n",
      "-rw-r--r--   1 neba  staff      794  2 Jun 17:56 fda913.pdb\r\n",
      "-rw-r--r--   1 neba  staff     1663  2 Jun 17:56 fda914.pdb\r\n",
      "-rw-r--r--   1 neba  staff     1663  2 Jun 17:56 fda915.pdb\r\n",
      "-rw-r--r--   1 neba  staff     1979  2 Jun 17:56 fda916.pdb\r\n",
      "-rw-r--r--   1 neba  staff      557  2 Jun 17:56 fda917.pdb\r\n",
      "-rw-r--r--   1 neba  staff     9875  2 Jun 20:14 p1_108.pdb\r\n",
      "-rw-r--r--   1 neba  staff    11087  2 Jun 20:14 p1_205.pdb\r\n",
      "-rw-r--r--   1 neba  staff     8057  2 Jun 20:14 p1_427.pdb\r\n",
      "-rw-r--r--   1 neba  staff     8855  2 Jun 20:14 p1_470.pdb\r\n",
      "-rw-r--r--   1 neba  staff     7865  2 Jun 20:14 p1_476.pdb\r\n",
      "-rw-r--r--   1 neba  staff     9582  2 Jun 20:14 p1_610.pdb\r\n",
      "-rw-r--r--   1 neba  staff     9774  2 Jun 20:14 p1_650.pdb\r\n",
      "-rw-r--r--   1 neba  staff     8259  2 Jun 20:14 p1_727.pdb\r\n",
      "-rw-r--r--   1 neba  staff     8875  2 Jun 20:14 p1_783.pdb\r\n",
      "-rw-r--r--@  1 neba  staff     7956  2 Jun 20:14 p1_79.pdb\r\n",
      "-rw-r--r--   1 neba  staff     9259  2 Jun 20:14 p1_859.pdb\r\n",
      "drwxr-xr-x@ 14 neba  staff      448  4 Jun 14:41 \u001b[34mpbmc_1k_v3_fastqs\u001b[m\u001b[m\r\n",
      "-rw-r--r--@  1 neba  staff       29  4 Jun 14:05 pbmc_1k_v3_fastqs.tar\r\n",
      "-rw-r--r--   1 neba  staff  5765121  3 Jun 03:56 rmsd700-900.csv\r\n",
      "-rw-r--r--   1 neba  staff  1709925  2 Jun 23:42 rmsd_p1-00_p1-Z_fda100.csv\r\n",
      "-rw-r--r--   1 neba  staff  1734226  2 Jun 23:55 rmsd_p1-0_p1-Z_fda200.csv\r\n",
      "-rw-r--r--   1 neba  staff  1734395  3 Jun 01:01 rmsd_p1-0_p1-Z_fda400.csv\r\n",
      "-rw-r--r--   1 neba  staff  1737059  3 Jun 01:13 rmsd_p1-0_p1-Z_fda500.csv\r\n",
      "-rw-r--r--   1 neba  staff  1204749  2 Jun 23:23 rmsd_p1-79_fda100.csv\r\n",
      "-rw-r--r--   1 neba  staff  1226121  3 Jun 00:13 rmsd_p1-79_fda200.csv\r\n",
      "-rw-r--r--   1 neba  staff  1226167  3 Jun 00:43 rmsd_p1-79_fda400.csv\r\n",
      "-rw-r--r--   1 neba  staff  1228954  3 Jun 01:31 rmsd_p1-79_fda500.csv\r\n",
      "-rw-r--r--   1 neba  staff  3469931  3 Jun 01:52 rmsd_p1_fda600.csv\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lG\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0c208dee",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Cannot find license file.\n",
      " The license files (or license server system network addresses) attempted are \n",
      "listed below.  Use LM_LICENSE_FILE to use a different license file,\n",
      " or contact your software provider for a license file.\n",
      "Feature:       PYMOL_MAIN\n",
      "Filename:      /Library/Application Support/Schrodinger/licenses\n",
      "License path:  /Library/Application Support/Schrodinger/licenses:\n",
      "FlexNet Licensing error:-1,359.  System Error: 2 \"No such file or directory\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(0.0, 0.0, 8, 5.0, 'control_fda1', 'fda1'), (0.0, 0.0, 38, 5.0, 'control_cyc1', 'cyclica1'), (0.0, 0.0, 57, 5.0, 'control_cyc2', 'cyclica2'), (0.0, 0.0, 69, 5.0, 'control_cyc3', 'cyclica3'), (0.0, 0.0, 182904945, 0.0, 'cyclica1', 'cyclica2'), (2.26, 2.26, 5, 0.0, 'control_cyc1', 'control_fda1'), (2.26, 2.26, 5, 0.0, 'control_cyc1', 'fda1'), (2.26, 2.26, 5, 0.0, 'control_fda1', 'cyclica1'), (2.26, 2.26, 5, 0.0, 'cyclica1', 'fda1'), (3.3, 3.3, 31, 5.0, 'control_cyc1', 'control_cyc3'), (3.3, 3.3, 31, 5.0, 'control_cyc1', 'cyclica3'), (3.3, 3.3, 31, 5.0, 'control_cyc3', 'cyclica1'), (3.3, 3.3, 31, 5.0, 'cyclica1', 'cyclica3'), (3.32, 3.32, 30, 5.0, 'control_cyc1', 'control_cyc2'), (3.32, 3.32, 30, 5.0, 'control_cyc1', 'cyclica2'), (3.32, 3.32, 30, 5.0, 'control_cyc2', 'cyclica1'), (3.39, 3.39, 7, 0.0, 'control_cyc2', 'control_fda1'), (3.39, 3.39, 7, 0.0, 'control_cyc2', 'fda1'), (3.39, 3.39, 7, 0.0, 'control_fda1', 'cyclica2'), (3.39, 3.39, 7, 0.0, 'cyclica2', 'fda1'), (3.42, 3.42, 45, 5.0, 'control_cyc2', 'control_cyc3'), (3.42, 3.42, 45, 5.0, 'control_cyc2', 'cyclica3'), (3.42, 3.42, 45, 5.0, 'control_cyc3', 'cyclica2'), (3.42, 3.42, 45, 5.0, 'cyclica2', 'cyclica3'), (3.99, 3.99, 6, 0.0, 'control_cyc3', 'control_fda1'), (3.99, 3.99, 6, 0.0, 'control_cyc3', 'fda1'), (3.99, 3.99, 6, 0.0, 'control_fda1', 'cyclica3'), (3.99, 3.99, 6, 0.0, 'cyclica3', 'fda1')]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RMSD_BR</th>\n",
       "      <th>RMSD_AR</th>\n",
       "      <th>#atomsAligned</th>\n",
       "      <th>Alignment_score</th>\n",
       "      <th>Ref_Ligand</th>\n",
       "      <th>Aligned_Ligand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>8</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_fda1</td>\n",
       "      <td>fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>38</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc1</td>\n",
       "      <td>cyclica1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>57</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc2</td>\n",
       "      <td>cyclica2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>69</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc3</td>\n",
       "      <td>cyclica3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>182904945</td>\n",
       "      <td>0.0</td>\n",
       "      <td>cyclica1</td>\n",
       "      <td>cyclica2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.26</td>\n",
       "      <td>2.26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_cyc1</td>\n",
       "      <td>control_fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2.26</td>\n",
       "      <td>2.26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_cyc1</td>\n",
       "      <td>fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2.26</td>\n",
       "      <td>2.26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_fda1</td>\n",
       "      <td>cyclica1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2.26</td>\n",
       "      <td>2.26</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>cyclica1</td>\n",
       "      <td>fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>3.30</td>\n",
       "      <td>3.30</td>\n",
       "      <td>31</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc1</td>\n",
       "      <td>control_cyc3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3.30</td>\n",
       "      <td>3.30</td>\n",
       "      <td>31</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc1</td>\n",
       "      <td>cyclica3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3.30</td>\n",
       "      <td>3.30</td>\n",
       "      <td>31</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc3</td>\n",
       "      <td>cyclica1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3.30</td>\n",
       "      <td>3.30</td>\n",
       "      <td>31</td>\n",
       "      <td>5.0</td>\n",
       "      <td>cyclica1</td>\n",
       "      <td>cyclica3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>3.32</td>\n",
       "      <td>3.32</td>\n",
       "      <td>30</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc1</td>\n",
       "      <td>control_cyc2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3.32</td>\n",
       "      <td>3.32</td>\n",
       "      <td>30</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc1</td>\n",
       "      <td>cyclica2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>3.32</td>\n",
       "      <td>3.32</td>\n",
       "      <td>30</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc2</td>\n",
       "      <td>cyclica1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>3.39</td>\n",
       "      <td>3.39</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_cyc2</td>\n",
       "      <td>control_fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>3.39</td>\n",
       "      <td>3.39</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_cyc2</td>\n",
       "      <td>fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>3.39</td>\n",
       "      <td>3.39</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_fda1</td>\n",
       "      <td>cyclica2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>3.39</td>\n",
       "      <td>3.39</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>cyclica2</td>\n",
       "      <td>fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>3.42</td>\n",
       "      <td>3.42</td>\n",
       "      <td>45</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc2</td>\n",
       "      <td>control_cyc3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>3.42</td>\n",
       "      <td>3.42</td>\n",
       "      <td>45</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc2</td>\n",
       "      <td>cyclica3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>3.42</td>\n",
       "      <td>3.42</td>\n",
       "      <td>45</td>\n",
       "      <td>5.0</td>\n",
       "      <td>control_cyc3</td>\n",
       "      <td>cyclica2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3.42</td>\n",
       "      <td>3.42</td>\n",
       "      <td>45</td>\n",
       "      <td>5.0</td>\n",
       "      <td>cyclica2</td>\n",
       "      <td>cyclica3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>3.99</td>\n",
       "      <td>3.99</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_cyc3</td>\n",
       "      <td>control_fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>3.99</td>\n",
       "      <td>3.99</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_cyc3</td>\n",
       "      <td>fda1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>3.99</td>\n",
       "      <td>3.99</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>control_fda1</td>\n",
       "      <td>cyclica3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3.99</td>\n",
       "      <td>3.99</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>cyclica3</td>\n",
       "      <td>fda1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    RMSD_BR  RMSD_AR  #atomsAligned  Alignment_score    Ref_Ligand  \\\n",
       "0      0.00     0.00              8              5.0  control_fda1   \n",
       "1      0.00     0.00             38              5.0  control_cyc1   \n",
       "2      0.00     0.00             57              5.0  control_cyc2   \n",
       "3      0.00     0.00             69              5.0  control_cyc3   \n",
       "4      0.00     0.00      182904945              0.0      cyclica1   \n",
       "5      2.26     2.26              5              0.0  control_cyc1   \n",
       "6      2.26     2.26              5              0.0  control_cyc1   \n",
       "7      2.26     2.26              5              0.0  control_fda1   \n",
       "8      2.26     2.26              5              0.0      cyclica1   \n",
       "9      3.30     3.30             31              5.0  control_cyc1   \n",
       "10     3.30     3.30             31              5.0  control_cyc1   \n",
       "11     3.30     3.30             31              5.0  control_cyc3   \n",
       "12     3.30     3.30             31              5.0      cyclica1   \n",
       "13     3.32     3.32             30              5.0  control_cyc1   \n",
       "14     3.32     3.32             30              5.0  control_cyc1   \n",
       "15     3.32     3.32             30              5.0  control_cyc2   \n",
       "16     3.39     3.39              7              0.0  control_cyc2   \n",
       "17     3.39     3.39              7              0.0  control_cyc2   \n",
       "18     3.39     3.39              7              0.0  control_fda1   \n",
       "19     3.39     3.39              7              0.0      cyclica2   \n",
       "20     3.42     3.42             45              5.0  control_cyc2   \n",
       "21     3.42     3.42             45              5.0  control_cyc2   \n",
       "22     3.42     3.42             45              5.0  control_cyc3   \n",
       "23     3.42     3.42             45              5.0      cyclica2   \n",
       "24     3.99     3.99              6              0.0  control_cyc3   \n",
       "25     3.99     3.99              6              0.0  control_cyc3   \n",
       "26     3.99     3.99              6              0.0  control_fda1   \n",
       "27     3.99     3.99              6              0.0      cyclica3   \n",
       "\n",
       "   Aligned_Ligand  \n",
       "0            fda1  \n",
       "1        cyclica1  \n",
       "2        cyclica2  \n",
       "3        cyclica3  \n",
       "4        cyclica2  \n",
       "5    control_fda1  \n",
       "6            fda1  \n",
       "7        cyclica1  \n",
       "8            fda1  \n",
       "9    control_cyc3  \n",
       "10       cyclica3  \n",
       "11       cyclica1  \n",
       "12       cyclica3  \n",
       "13   control_cyc2  \n",
       "14       cyclica2  \n",
       "15       cyclica1  \n",
       "16   control_fda1  \n",
       "17           fda1  \n",
       "18       cyclica2  \n",
       "19           fda1  \n",
       "20   control_cyc3  \n",
       "21       cyclica3  \n",
       "22       cyclica2  \n",
       "23       cyclica3  \n",
       "24   control_fda1  \n",
       "25           fda1  \n",
       "26       cyclica3  \n",
       "27           fda1  "
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#!/usr/bin/python\n",
    "#\n",
    "# PairWise_RMSD_smalLigs.py\n",
    " \n",
    "import __main__\n",
    "__main__.pymol_argv = [ 'pymol', '-qc'] # Quiet and no GUI\n",
    " \n",
    "import sys, time, os, re\n",
    "from pymol import cmd\n",
    "import pymol\n",
    "import glob\n",
    "import pandas as pd\n",
    "\n",
    "pymol.finish_launching()\n",
    "\n",
    "directory = \"/Users/neba/Desktop/CyclicaVsFDA/Results/TEST_cyclica_FDA/fda1*.pdb\"\n",
    "#directory2 = \"/Users/neba/Desktop/CyclicaVsFDA/Results/test1/*.pdb\"\n",
    "\n",
    "\n",
    "# Read User Input\n",
    "structurePath = glob.glob(directory)\n",
    "#structurePath2 = glob.glob(directory2)\n",
    "\n",
    "structurePath.sort()\n",
    "#structurePath2.sort()\n",
    "extension = re.compile( '(^.*[\\/]|\\.(pdb|ent|brk))' )\n",
    "\n",
    "object_list = []\n",
    "for filename in structurePath:\n",
    "    object_list.append(extension.sub ('',filename))\n",
    "\n",
    "#for filename in structurePath2:\n",
    "    #object_list.append(extension.sub ('',filename))\n",
    "\n",
    "rmsd = {}    \n",
    "rmsd_list = []\n",
    "for i in range (len(structurePath)):\n",
    "    for j in range(1+i,len(structurePath)):\n",
    "        obj_name1 = extension.sub('',structurePath[i].split(\"/\")[-1].split(\".\")[0])\n",
    "        cmd.load(structurePath[i],obj_name1)\n",
    "        obj_name2 = extension.sub('',structurePath[j].split(\"/\")[-1].split(\".\")[0])\n",
    "        cmd.load(structurePath[j],obj_name2)\n",
    "        \n",
    "        rmsd_val = cmd.align(obj_name1, obj_name2, cycles=5, cutoff=5,\n",
    "                            quiet=1, transform=0)#good ligand alignment when transform=0\n",
    "        #transform=0/1: do superposition {default: 1}\n",
    "                             \n",
    "        \n",
    "        rmsd.setdefault(obj_name1,{})[obj_name2] = rmsd_val[0],rmsd_val[1],rmsd_val[2],\n",
    "        rmsd_val[3], rmsd_val[4],rmsd_val[5], rmsd_val[6]\n",
    "        \n",
    "        rmsd_list.append((round(rmsd_val[3],2), round(rmsd_val[0],2), round(rmsd_val[4],2),\n",
    "                          round(rmsd_val[5],2), obj_name1, obj_name2)) #indexing to creat\n",
    "                                                                          #columns in df\n",
    "             #columns in df\n",
    "        \n",
    "        #Delete object and replace with new items when \n",
    "        #loop loops through directory, the delete ccommnds must\n",
    "        #align with loops for good results\n",
    "       \n",
    "        cmd.delete(obj_name2)\n",
    "    cmd.delete(obj_name1)\n",
    "\n",
    "#Re-order table\n",
    "rmsd_list.sort()\n",
    "\n",
    "#testing\n",
    "print (rmsd_list)\n",
    "\n",
    "#Create tables with poutput pairwise rmsds\n",
    "rmsd_df = pd.DataFrame(rmsd_list)\n",
    "\n",
    "#Renaming columns to match pymol's \"align\" column names\n",
    "rmsd_df.columns = ['RMSD_BR','RMSD_AR', '#atomsAligned',\n",
    "                   'Alignment_score','Ref_Ligand', \n",
    "                   'Aligned_Ligand']\n",
    "\n",
    "#save file to directory\n",
    "rmsd_df.to_csv(\"rmsd900-1000_test.csv\", index=False)\n",
    "\n",
    "rmsd_df.head(50)\n",
    "\n",
    "#The \"align\" tool of pymol returns a list with 7 items:\n",
    "#RMSD after refinement\n",
    "#Number of aligned atoms after refinement\n",
    "#Number of refinement cycles\n",
    "#RMSD before refinement\n",
    "#Number of aligned atoms before refinement\n",
    "#Raw alignment score\n",
    "#Number of residues aligned\n",
    "#\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa7e861b",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.read_csv('rmsd900-1000.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fcae5b8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "rmsd_df[(rmsd_df[\"RMSD_BR\"] < 2) & (rmsd_df[\"Alignment_score\"] > 3)\n",
    "        & (rmsd_df['#atomsAligned'] > 5)].head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6f4ea9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#import time for timing runs\n",
    "import time\n",
    "start_time = time.time()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9db75a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import scanpy as sc\n",
    "import anndata\n",
    "from sklearn.decomposition import TruncatedSVD\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c02323f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# install kb\n",
    "!pip install --quiet kb-python "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7bcd6571",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# Download the data from the 10x website\n",
    "!curl -O http://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v3/pbmc_1k_v3_fastqs.tar\n",
    "\n",
    "# unpack the downloaded files\n",
    "!tar -xf pbmc_1k_v3_fastqs.tar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "028741e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "!tar -czf pbmc_1k_v3_fastqs.tar pbmc_1k_v3_fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6222147b",
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls *.tar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd7487c0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
